src/kOS.Safe/Screen/ScreenBuffer.cs

```diff
@@ -258,6 +296,27 @@ public List<char[]> GetBuffer()
             return mergedBuffer;
         }
 
+        // This was handy when trying to figure out what was going on.  
+        public string DebugDump()
+        {
+            StringBuilder sb = new StringBuilder();
+            sb.Append("DebugDump ScreenBuffer: RowCount="+RowCount+", ColumnCount="+ColumnCount+", topRow="+topRow+", buffer.count="+buffer.Count+"\n");
+            for (int i = 0; i < buffer.Count ; ++i)
+            {
+                sb.Append(" line "+i+" = [");
+                for (int j = 0 ; j < buffer[i].Length ; ++j)
+                {
+                    char ch = buffer[i][j];
+                    sb.Append((int)ch < 32 ? (" \\"+(int)ch) : (" "+ch) );
+                }
+                sb.Append("]\n");
+            }
+            foreach (SubBuffer sub in subBuffers)
+                sb.Append(sub.DebugDump());
+            return sb.ToString();
+        }
```

Comment:
This isn't being used in the checked-in code because making use of it spams the output log and slows KSP down.  But it was very handy for debugging things and I'd like it left in place so it's still available for any future expansions to be able to debug with it.